### PR TITLE
Add layerId to AnnotationManager

### PR DESCRIPTION
**Pull request by julianrex**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [X] Briefly describe the changes in this PR.
 - [ ] ~Include before/after visuals or gifs if this PR includes visual changes.~
 - [X] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] ~Add example if relevant.~
 - [X] Document any changes to public APIs.
 - [X] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [X] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog>Added function to get layer identifier for an annotation type.</changelog>`.
 - [ ] ~Update the migration guide, API Docs, Markdown files - Readme, Developing, etc~

### Summary of changes

This PR adds a function to get the layer id for a particular annotation type. It also adds a WIP convenience function (`_moveLayer`) to `Style` to allow layers to be moved. These combined give developers a method to move the underlying annotation layer. 

The `_moveLayer` change is marked as `:nodoc:` and should be consider prone to change.

cc @Neel

### CODE REVIEW

**Comment by julianrex** [comment]
cc @Chaoba @MaximAlien @jill-cardamon

File: Sources/MapboxMaps/Annotations/Configuration/AnnotationOptions.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/// :nodoc:
+public struct AnnotationSourceOptions: Equatable {
+    // TODO: Add source and cluster options
+}
+
+/// Configuration options for the AnnotationManager
+public struct AnnotationOptions: Equatable {
+    /// The identifier of the layer above the annotation layers. Default is nil, which will position the layers
+    /// above existing layers.
+    public var belowLayerId: String?
```
Comment (neelmistry94) [inline Sources/MapboxMaps/Annotations/Configuration/AnnotationOptions.swift:12]: why are we only account for below and not above?

Comment (julianrex) [reply Sources/MapboxMaps/Annotations/Configuration/AnnotationOptions.swift:12]: This is to align with Android - but agreed, this should probably use `LayerPosition` for both platforms.

Comment (neelmistry94) [reply Sources/MapboxMaps/Annotations/Configuration/AnnotationOptions.swift:12]: can we ticket that out then. I'm concerned that we only provide the one option

**Review by neelmistry94 — APPROVED** [review]

File: Sources/MapboxMaps/Annotations/Configuration/AnnotationOptions.swift (hunk 1 above)
Comment (julianrex) [reply Sources/MapboxMaps/Annotations/Configuration/AnnotationOptions.swift]: I changed layer position handling in a subsequent commit. Will ticket out difference.

**Review by neelmistry94 — APPROVED** [review]

**Pull request merged**